platform/darwin/src/MGLLoggingConfiguration.mm

```diff
@@ -1,14 +1,43 @@
+#include <mbgl/util/logging.hpp>
+
 #import "MGLLoggingConfiguration_Private.h"
 
 #ifndef MGL_LOGGING_DISABLED
 #if __has_builtin(__builtin_os_log_format)
 #import <os/log.h>
 #endif
 
-@implementation MGLLoggingConfiguration
+namespace mbgl {
+    
+class MGLCoreLoggingObserver : public Log :: Observer {
+public:
+    //Return true not print messages at core level, and filter at platform level.
+    bool onRecord(EventSeverity severity, Event, int64_t, const std::string& msg) override{
+        switch (severity) {
+            case EventSeverity::Debug:
+                MGLLogDebug([NSString stringWithCString:msg.c_str() encoding:NSUTF8StringEncoding]);
```

Comment:
Let's include the event and code in the `NSString` too - these will help locate the error message.